core/src/transport/memory.rs

```diff
@@ -418,28 +455,25 @@ mod tests {
     #[test]
     fn listening_twice() {
         let mut transport = MemoryTransport::default();
-        assert!(transport
-            .listen_on("/memory/1639174018481".parse().unwrap())
-            .is_ok());
-        assert!(transport
-            .listen_on("/memory/1639174018481".parse().unwrap())
-            .is_ok());
-        let _listener = transport
-            .listen_on("/memory/1639174018481".parse().unwrap())
-            .unwrap();
-        assert!(transport
-            .listen_on("/memory/1639174018481".parse().unwrap())
-            .is_err());
-        assert!(transport
-            .listen_on("/memory/1639174018481".parse().unwrap())
-            .is_err());
-        drop(_listener);
-        assert!(transport
-            .listen_on("/memory/1639174018481".parse().unwrap())
-            .is_ok());
-        assert!(transport
-            .listen_on("/memory/1639174018481".parse().unwrap())
-            .is_ok());
+
+        let addr_1: Multiaddr = "/memory/1639174018481".parse().unwrap();
+        let addr_2: Multiaddr = "/memory/8459375923478".parse().unwrap();
+
+        let listener_id_1 = transport.listen_on(addr_1.clone()).unwrap();
+        assert!(transport.remove_listener(listener_id_1));
```

Comment:
Perhaps we could add panic messages here to make it a bit easier to understand, what is being asserted.

Reply:
Done in b19e11a.